addons/account/models/account_journal.py

```diff
@@ -502,6 +503,10 @@ def _check_auto_post_draft_entries(self):
     @api.onchange('type')
```

Comment:
I wonder if an api.depends or even a python/SQL constraint would be appropriate for either (refund and payment sequence) settings.